test/src/main/java/org/apache/accumulo/test/functional/ConcurrentTableNameOperationsIT.java

```diff
@@ -0,0 +1,414 @@
+          }
+        }));
+
+        futures.add(pool.submit(() -> {
+          try {
+            startSignal.countDown();
+            startSignal.await();
+            client.tableOperations().clone(cloneSourceTables.get(index), targetTableName, true,
+                Map.of(), Set.of());
+            successfulOperation.set("clone");
+            return true;
+          } catch (TableExistsException e) {
+            return false;
+          }
+        }));
+      }
+
+      int successCount = 0;
+      for (Future<Boolean> future : futures) {
+        if (future.get()) {
+          successCount++;
+        }
+      }
+
+      assertEquals(1, successCount, "Expected only one operation to succeed");
+
+      int tableCountAfter = client.tableOperations().list().size();
+      assertTrue(client.tableOperations().exists(targetTableName),
+          "Expected target table " + targetTableName + " to exist");
+
+      String operation = successfulOperation.get();
+      if ("create".equals(operation) || "clone".equals(operation)) {
+        assertEquals(tableCountBefore + 1, tableCountAfter,
+            "Expected +1 table count for " + operation);
+      } else if ("rename".equals(operation)) {
+        assertEquals(tableCountBefore, tableCountAfter, "Expected same table count for rename");
+      }
+    }
+
+    pool.shutdown();
+  }
+
+  /**
+   * Test that when several threads attempt to create the same namespace simultaneously, only one
+   * creation succeeds.
+   */
+  @Test
+  public void createNamespace() throws Exception {
+    final int numTasks = 10;
+    final int numIterations = 3;
+    ExecutorService pool = Executors.newFixedThreadPool(numTasks);
+    String[] targetNamespaceNames = getUniqueNames(numIterations);
+
+    for (String namespaceName : targetNamespaceNames) {
+      Set<String> namespacesBefore = client.namespaceOperations().list();
+
+      int successCount = runConcurrentNamespaceOperation(pool, numTasks, (index) -> {
+        client.namespaceOperations().create(namespaceName);
+        return true;
+      });
+
+      assertEquals(1, successCount, "Expected only one create operation to succeed");
+      assertTrue(client.namespaceOperations().exists(namespaceName),
+          "Expected namespace " + namespaceName + " to exist");
+
+      Set<String> namespacesAfter = client.namespaceOperations().list();
+      Set<String> newNamespaces = new HashSet<>(namespacesAfter);
+      newNamespaces.removeAll(namespacesBefore);
+      assertEquals(Set.of(namespaceName), newNamespaces,
+          "Expected exactly one new namespace: " + namespaceName);
+
+      client.namespaceOperations().delete(namespaceName);
+    }
+
+    pool.shutdown();
+  }
+
+  /**
+   * Test that when several threads attempt to rename to the same namespace name simultaneously,
+   * only one rename succeeds.
+   */
+  @Test
+  public void renameNamespace() throws Exception {
+    final int numTasks = 10;
+    final int numIterations = 3;
```

Comment:
For some test `numIterations` is 10 and for others it is 3, was this adjusted based on the test run times?

Reply:
Yea based on run times. I made some slight adjustments so they all run in ~30 seconds on my machine in commit 5a7deeb